src/cargo/ops/resolve.rs

```diff
@@ -171,6 +171,32 @@ pub fn resolve_ws_with_opts<'cfg>(
         feature_opts,
     )?;
 
+    let no_lib_warnings: Vec<String> = pkg_set
+        .no_lib_pkgs(
+            &resolved_with_overrides,
+            &member_ids,
+            has_dev_units,
+            requested_targets,
+            target_data,
+            force_all_targets,
+        )
+        .iter()
+        .flat_map(|(pkg_id, dep_pkgs)| {
+            dep_pkgs.iter().map(move |dep_pkg| {
+                format!(
+                    "{} has invalid dependency `{}`. `{}` has no lib package.",
```

Comment:
The word `package` here is a bit confusing, since a "lib" is usually called a target.  Also, style-wise we don't end messages with a `.`, and repeating the dependency name twice appears a little bit as a stutter.  Perhaps something like this would work?  

```suggestion
                    "{} ignoring invalid dependency `{}` which is missing a lib target
```

It's a bit tricky to word it clearly an succinctly.

Reply:
Addressed. Thanks!